### ISSUE TITLE
GET points/<id> != POST points { ids: [<ids>] }

### ISSUE THREAD

**Issue by hdmi** [qdrant/qdrant#5071]
There are two ways of retrieving points from a collection

### Single retrieve
`GET /collections/{collection_name}/points/{point_id}`

### Multi retrieve
```
POST /collections/{collection_name}/points
{
    "ids": [0, 3, 100]
}

```
Those two functions retrieve me different things.

## Current Behavior

Lets create a custom point with ID 2637283987631399372 (fits in an unsigned long of 64 bits)

```
PUT /collections/my_collection/points
{
    "points": [
        {
            "id": 2637283987631399372,
            "payload": {"color": "red"},
            "vector": [0.1545465, ... ,-0.00987415]
        }
    ]
}
```

Created OK, then if I call

`GET /collections/my_collection/points/2637283987631399372`

I get

```
{
  "error": "Not found: Point with id 2637283987631399372 does not exists!"
}
```

umm... strange, lets call

```
POST /collections/my_collection/points
{
    "ids": [2637283987631399372 ]
}

```
I get

```
{
  "result": [
    {
      "id": 2637283987631399400
    }
  ],
  "status": "ok",
  "time": 0.0001612
}
```

## My questions
Why the two methods of retrieving points behave differently? Is there anything that I am missing? 

By the way has the server just rounded the ID that I have just provided?
My ID 2637283987631399**372**, the ID I get back 2637283987631399**400**

Running the server in v1.11.3 Windows MSCV

## Solution

JSON does not allow BigInts, so the 64-bit IDs from qdrant might be truncated when using big numbers inside a JSON that needs to be sent to the server.

**Comment by timvisee** [comment]
Nice catch! Sadly this is caused by how JSON behaves in JavaScript. JavaScript doesn't have proper 64 bit numbers, but rather [53 bits](https://stackoverflow.com/a/9643650/1000145).

This is a problem because it cannot represent large point IDs in JSON objects. Instead it truncates the number and rounds it to the nearest number it can represent.

The behavior of the two endpoints appears different because one uses JSON, the other does not.

Please note that this only happens in JavaScript. None of our other clients/interfaces have this problem and work as expected.

We have made efforts to minimize this problem. But as far as I know it's not 100% fixable due to JavaScript limitations.

**Comment by hdmi** [comment]
Oh, I see. Now that you mentioned I inspected the stringified JSON in the request and the ID was being truncated as you said.

I worked around it by manually stringifying the JSON with something like...

```js
// Manual JSON stringify to POST
const payload_stringified = `{"ids":[${idsArray.join(',')}], "with_payload": true}`;
```

Certainly much quicker than issuing N individual point requests.

And also thanks to the server side implementation that parses the big numbers from the non-compliant JSON as expected.
